src/virtualenvrunner/pythonversionrun.py

```diff
@@ -20,16 +20,22 @@ def _get_run_lambda(self, python_version):
         return lambda: self._run_with_python_version(python_version)
 
     def _run_with_python_version(self, python_version):
-        self._run(pythonexe='python{major}.{minor}{exe_suffix}'.format(
-            major=python_version.major,
-            minor=python_version.minor,
-            exe_suffix=get_exe_suffix()),
-                  python_version=python_version)
+        if python_version.minor:
+            self._run(pythonexe='python{major}.{minor}{exe_suffix}'.format(
+                major=python_version.major,
+                minor=python_version.minor,
+                exe_suffix=get_exe_suffix()),
+                    python_version=python_version)
+        else:
+            self._run(pythonexe='python{major}{exe_suffix}'.format(
+                major=python_version.major,
+                exe_suffix=get_exe_suffix()),
+                python_version=python_version)
 
     def __getattr__(self, name):
         try:
             return self._run_functions[name]
         except KeyError:
-            raise AttributeError('{cls} object has no attribute {name}'.format(
+            raise AttributeError('{cls} object has no attribute {name}'.format(  # pylint: disable=raise-missing-from
```

Comment:
Disabling pylint checks `raise-missing-from` and `super-with-arguments` should be on its own commit, as it is not related to content of this change.

Reply:
Added as new commit

Reply:
OK